### PR TITLE
add support for new ShoppingCart and CartItem entities

### PR DESCRIPTION
**Pull request by AntonKurylo**

### CODE REVIEW

File: src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+                .filter(item -> item.getBook().getId().equals(requestDto.bookId()))
+                .findFirst()
+                .map(item -> {
+                    item.setQuantity(item.getQuantity() + requestDto.quantity());
+                    return item;
+                }).orElse(shoppingCartMapper.toCartItemEntity(requestDto, shoppingCart));
+        shoppingCart.getCartItems().add(cartItem);
+        shoppingCartRepository.save(shoppingCart);
+        return shoppingCartMapper.toShoppingCartDto(
+                shoppingCartRepository.findByUserId(authenticatedUserId));
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java]: Could you avoid calling shoppingCartRepository.findByUserId twice?

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
one comment

File: src/main/java/mate/academy/repository/cartitem/CartItemRepository.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package mate.academy.repository.cartitem;
+
+import mate.academy.model.CartItem;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
```
Comment (okuzan) [inline src/main/java/mate/academy/repository/cartitem/CartItemRepository.java:7]: not a mistake but remember that this annotation is optional when we extend JpaRepository

File: src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+                .filter(item -> item.getBook().getId().equals(requestDto.bookId()))
+                .findFirst()
+                .map(item -> {
+                    item.setQuantity(item.getQuantity() + requestDto.quantity());
+                    return item;
+                }).orElseGet(() -> {
+                    CartItem entity = shoppingCartMapper.toCartItemEntity(requestDto, shoppingCart);
+                    entity.setBook(book);
+                    return entity;
+                });
```
Comment (okuzan) [inline src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java:59]: a bit unwieldy, i would consider extracting to a private method

**Review by okuzan — APPROVED** [review]

**Pull request merged**